File: python_Programmers/1단계/약수의합.py
# 내 풀이
def solution(n):
    answer = 0
    for i in range(1, n+1):
        if n % i == 0:
            answer += i
    return answer

# 모범풀이
def solution2(num):
    # num / 2 의 수들만 검사하면 성능 약 2배 향상잼
    return num + sum([i for i in range(1, (num // 2) + 1) if num % i == 0])


n = 5
print(solution(n))
print(solution2(n))

File: java/programmers/1단계/src/약수의합.java
public class 약수의합 {
    public int solution(int n) {
        int answer = 0;
        // n의 절반은 n*2 그러므로 그위의 값은 자기자신을 제외하고 약수가 아님
        for (int i = 1; i <= n/2 ; i++) {
            if(n % i == 0) answer+= i;
        }
        return answer + n;
    }

    public static void main(String[] args) {
        약수의합 sol = new 약수의합();
        int n = 12;
        System.out.println(sol.solution(n));
    }
}
